src/snapshot.ts

```diff
@@ -85,85 +86,17 @@ import * as util from './util';
  * });
  */
 export class Snapshot {
-  parent;
+  parent: Subscription|PubSub;
   name: string;
   // tslint:disable-next-line variable-name
   Promise?: PromiseConstructor;
-  create;
-  seek;
-  metadata;
-  constructor(parent, name: string) {
-    if (parent.Promise) {
+  metadata!: google.pubsub.v1.Snapshot;
+  constructor(parent: Subscription|PubSub, name: string) {
+    if (parent instanceof PubSub) {
       this.Promise = parent.Promise;
     }
     this.parent = parent;
     this.name = Snapshot.formatName_(parent.projectId, name);
-    if (is.fn(parent.createSnapshot)) {
```

Comment:
Feels like this requires a bit of an explanation :) What's up here?

Reply:
For whatever reason the PubSub API allows you to list all snapshots for a project without providing information about the parent subscription. This gets kind of funky because you need to know the subscription name in order to create or seek a snapshot. In the event that the user got a snapshot via `PubSub#snapshot` or `PubSub#getSnapshots` then `create` and `seek` simply would not exist. The changes here just make it so that instead you can call those methods, but they will return an error stating why they aren't available.